### PR TITLE
Fix RESTEasy Reactive Default Codestart

### LINKED ISSUE #16497: `quarkus:create` adds `quarkus-resteasy` to a project even when `quarkus-resteasy-reactive-jackson` is specified
## Describe the bug
When generating a project with `quarkus-resteasy-reactive-jackson`, `quarkus-resteasy` extension is added to the project's pom.

Related to https://github.com/quarkusio/quarkus/issues/15997

### Expected behavior
`quarkus-resteasy-reactive` and `quarkus-resteasy` are mutually exclusive, the latter should not be added when the former is used, and `quarkus-resteasy-reactive-jackson`  brings in the former.

### Actual behavior
The dependency on `quarkus-resteasy` is added. 
It all works okay if a user explicitly adds `quarkus-resteasy-reactive`

## To Reproduce
```
mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create -Dextensions=resteasy-reactive-jackson
```

### PR DESCRIPTION
**Pull request by ia3andy**
Fixes #16497

### CODE REVIEW

**Review by geoand — APPROVED** [review]

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-jsonb/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 1]
```diff
@@ -1,4 +1,3 @@
----
```
Comment (gsmet) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive-jsonb/runtime/src/main/resources/META-INF/quarkus-extension.yaml:1]: Given `yamllint` asks for this to be present by default, I wouldn't actively remove it. I won't block this PR for this but let's not remove them in others please.

Comment (ia3andy) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-jsonb/runtime/src/main/resources/META-INF/quarkus-extension.yaml:1]: @gsmet there are meant for multi documents yaml file which is not used here, maybe yamllint needs some setting and not the other way around..

**Comment by gsmet** [comment]
This is not backportable as is:
```
 Error:  Failed to execute goal io.quarkus:quarkus-bootstrap-maven-plugin:999-SNAPSHOT:extension-descriptor (generate-extension-descriptor) on project quarkus-resteasy-reactive-jackson: Failed to resolve codestart artifact io.quarkus:quarkus-project-core-extension-codestarts:jar:999-SNAPSHOT: Failed to resolve io.quarkus:quarkus-project-core-extension-codestarts:jar:999-SNAPSHOT: Failed to resolve artifact io.quarkus:quarkus-project-core-extension-codestarts:jar:999-SNAPSHOT: Could not find artifact io.quarkus:quarkus-project-core-extension-codestarts:jar:999-SNAPSHOT -> [Help 1]
```

@ia3andy can you have a look?

**Comment by ia3andy** [comment]
@gsmet I need to make a special fix for 1.13

**Comment by ia3andy** [comment]
done: https://github.com/quarkusio/quarkus/pull/16824

**Pull request merged**